### ISSUE TITLE
Weird case of tokenization when dealing with hyphens

### ISSUE THREAD

**Issue by parthjoshi2007** [explosion/spaCy#2726]
I've been scratching my head over the following case of tokenization in case of hyphens:

```
US-based  => ['US', '-', 'based'] (correct)
US- based => ['US-', 'based'] (wrong)
```

Notice that the additional space in the middle of the hyphenated word causes the hyphen to not be recognized as a separate token. This case occurs in my dataset often enough that I need to figure out how I can fix the tokenization, particularly for cases where the token is part of a named entity, as above.
Going further, I would assume that the default behaviour for the tokenizer should be that any punctuation at the end of a token should be split into its own separate token except for . in the case of abbreviations like U.S.

**UPDATE**: I just noticed that for most other punctuation the behaviour is as expected; this incorrect tokenization happens specifically for hyphens.

Is there a simple fix to my problem? I am fairly new to SpaCy. Thanks

## Environment

* Operating System: Ubuntu 18..04
* Python Version Used: 3.6
* spaCy Version Used: 2.0.12

**Comment by parthjoshi2007** [comment]
I've managed to get it working for now by modifying `spacy.lang.punctuation.TOKENIZER_SUFFIXES` as follows:-
```
DASHES = ['-'] + [chr(i) for i in range(8208, 8214)]
TOKENIZER_SUFFIXES.extend(DASHES)
```
But I still have a remaining query. I see that there is a method to merge tokens within a span. Is there a way to split tokens that are created? I could not find any way to do so. This would be ideal for me since I have annotated text where my annotations define the canonical token boundaries but these sometimes do not align with SpaCy's token boundaries (in cases of ill-formed sentences with missing whitespace etc). Any help is appreciated. Thanks.
